CONTRIBUTING.md

```diff
@@ -48,46 +51,131 @@ Once you have cloned the [GitHub repository](https://github.com/datastrato/gravi
+- Type and select `Shell Command: Install 'code' command in PATH`.
+
+3. **Ensure Remote - WSL is Active**
+   When VSCode opens, you should see a green bottom-left corner indicating that VSCode is connected to WSL. If it isn't, click on the green area and select `Remote-WSL: New Window` or `Remote-WSL: Reopen Folder in WSL`.
+
+4. **Edit and Develop Your Project**
+   You can now edit and develop your project files in VSCode as if they were local files. The Remote - WSL extension seamlessly bridges the file system between Windows and WSL.
+
+### Handling Memory Issues in WSL
+
+If you ran into a memory issue when using WSL, here are some solutions to resolve it.
+
+1. **Shut down WSL**  
+   If your WSL is open, you can shut it down in Windows PowerShell using the following command:  
+   ```powershell
+   wsl --shutdown
+
+   ```
+
+2. **Navigate to user folder**  
+   Open up File Explorer and navigate to `C:\Users\<your-username>`.
 
+3. **Create the `.wslconfig` file**  
+   Open up Notepad or another text editor and input the following:  
+   ```plaintext
+   [wsl2]
+   memory=4GB   # Limits VM memory in WSL 2 up to 4GB
+   processors=4 # Makes the WSL 2 VM use four virtual processors
+   ```
+   *The memory and processor usage can be changed depending on your system's hardware.*
+
+4. **Save the file**  
+   Save the file as `".wslconfig"`. Be sure to include the quotes to let Windows know that this isn't a text file.
+
+## Project Overview and Policies
+### Project Overview
 For an overview of the project, see [README.md](README.md).
 
 ### Management Policies
-
 For project management policies, refer to [GOVERNANCE.md](GOVERNANCE.md).
 
 ### Future Development Directions
-
 For future development directions, refer to the [ROADMAP.md](ROADMAP.md) document.
 
 ## Contribution guidelines
-
-### Code of conduct
-
+### Code of Conduct
 Please read and follow the [Code of Conduct](CODE_OF_CONDUCT.md). Gravitino provides a welcoming and inclusive environment for all contributors.
-
 ### Reporting bugs
 
 If you find a bug in Gravitino, please open an issue on GitHub. Be sure to include as much detail as possible, such as a clear description, steps to reproduce, and your environment. Please follow the template provided. If you encounter a security issue, please refer to [SECURITY.md](SECURITY.md).
 
 ### Suggesting enhancements
-
 If you have ideas for enhancements or new features, feel free to create an issue to discuss them. Gravitino welcomes suggestions and provides prompt feedback on their feasibility and relevance.
 
-### Good first issues
-
-If you are new to open source or can't find something to work on check out the [Good First Issues list](https://github.com/datastrato/gravitino/contribute).
-
-### Working on issues
+### Good First Issues
+If you are new to open source or can't find something to work on, check out the [Good First Issues list](https://github.com/datastrato/gravitino/contribute).
```

Comment:
Again, you should not change this. You should leave blank line before and after for each paragraph. Please recheck all your changes and update the doc.

Reply:
Got it, done!